source/CMakeLists.txt

```diff
@@ -109,6 +115,22 @@ macro(spvtools_vendor_tables VENDOR_TABLE)
   set_property(TARGET spirv-tools-${VENDOR_TABLE} PROPERTY FOLDER "SPIRV-Tools build")
 endmacro(spvtools_vendor_tables)
 
+macro(spvtools_extinst_lang_headers NAME GRAMMAR_FILE)
+  set(OUTBASE ${spirv-tools_BINARY_DIR}/${NAME})
+  set(OUT_H ${OUTBASE}.h)
+  add_custom_command(OUTPUT ${OUT_H}
+    COMMAND ${PYTHON_EXECUTABLE} ${LANG_HEADER_PROCESSING_SCRIPT}
+      --extinst-name=${NAME}
+      --extinst-grammar=${GRAMMAR_FILE}
+      --extinst-output-base=${OUTBASE}
+    DEPENDS ${LANG_HEADER_PROCESSING_SCRIPT} ${GRAMMAR_FILE}
+    COMMENT "Generate language specific header for ${NAME}.")
+  list(APPEND EXTINST_CPP_DEPENDS ${OUT_H})
+  add_custom_target(spirv-tools-${NAME} DEPENDS ${OUT_H})
```

Comment:
Maybe use `spirv-tools-header-${NAME}` here?

I  see the names passed into `spvtools_vendor_tables()` and `spvtools_extinst_lang_headers()` only differ w.r.t. cases. If we provide the same name to the above two macros, targets will collide. It is subtle.